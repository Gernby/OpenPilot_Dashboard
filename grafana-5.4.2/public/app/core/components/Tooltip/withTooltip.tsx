import React from 'react';
import { Manager, Popper, Arrow } from 'react-popper';

interface IwithTooltipProps {
  placement?: string;
  content: string | ((props: any) => JSX.Element);
  className?: string;
}

export default function withTooltip(WrappedComponent) {
  return class extends React.Component<IwithTooltipProps, any> {
    constructor(props) {
      super(props);

      this.setState = this.setState.bind(this);
      this.state = {
        placement: this.props.placement || 'auto',
        show: false,
      };
    }

    componentWillReceiveProps(nextProps) {
      if (nextProps.placement && nextProps.placement !== this.state.placement) {
        this.setState(prevState => {
          return {
            ...prevState,
            placement: nextProps.placement,
          };
        });
      }
    }

    renderContent(content) {
      if (typeof content === 'function') {
        // If it's a function we assume it's a React component
        const ReactComponent = content;
        return <ReactComponent />;
      }
      return content;
    }

    render() {
      const { content, className } = this.props;

      return (
        <Manager className={`popper__manager ${className || ''}`}>
          <WrappedComponent {...this.props} tooltipSetState={this.setState} />
          {this.state.show ? (
            <Popper placement={this.state.placement} className="popper">
              {this.renderContent(content)}
              <Arrow className="popper__arrow" />
            </Popper>
          ) : null}
        </Manager>
      );
    }
  };
}
