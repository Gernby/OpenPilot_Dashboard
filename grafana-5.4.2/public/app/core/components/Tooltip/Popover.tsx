import React from 'react';
import withTooltip from './withTooltip';
import { Target } from 'react-popper';

interface PopoverProps {
  tooltipSetState: (prevState: object) => void;
}

class Popover extends React.Component<PopoverProps, any> {
  constructor(props) {
    super(props);
    this.toggleTooltip = this.toggleTooltip.bind(this);
  }

  toggleTooltip() {
    const { tooltipSetState } = this.props;
    tooltipSetState(prevState => {
      return {
        ...prevState,
        show: !prevState.show,
      };
    });
  }

  render() {
    return (
      <Target className="popper__target" onClick={this.toggleTooltip}>
        {this.props.children}
      </Target>
    );
  }
}

export default withTooltip(Popover);
