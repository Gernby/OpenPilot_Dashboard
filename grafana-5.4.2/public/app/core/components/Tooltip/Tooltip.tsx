import React, { PureComponent } from 'react';
import withTooltip from './withTooltip';
import { Target } from 'react-popper';

interface Props {
  tooltipSetState: (prevState: object) => void;
}

class Tooltip extends PureComponent<Props> {
  showTooltip = () => {
    const { tooltipSetState } = this.props;

    tooltipSetState(prevState => ({
      ...prevState,
      show: true,
    }));
  };

  hideTooltip = () => {
    const { tooltipSetState } = this.props;
    tooltipSetState(prevState => ({
      ...prevState,
      show: false,
    }));
  };

  render() {
    return (
      <Target className="popper__target" onMouseOver={this.showTooltip} onMouseOut={this.hideTooltip}>
        {this.props.children}
      </Target>
    );
  }
}

export default withTooltip(Tooltip);
