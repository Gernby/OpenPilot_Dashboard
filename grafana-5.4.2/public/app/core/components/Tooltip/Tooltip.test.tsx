import React from 'react';
import renderer from 'react-test-renderer';
import Tooltip from './Tooltip';

describe('Tooltip', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Tooltip className="test-class" placement="auto" content="Tooltip text">
          <a href="http://www.grafana.com">Link with tooltip</a>
        </Tooltip>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
