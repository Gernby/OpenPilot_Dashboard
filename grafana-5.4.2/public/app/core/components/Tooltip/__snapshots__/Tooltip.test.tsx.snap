// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders correctly 1`] = `
<div
  className="popper__manager test-class"
>
  <div
    className="popper__target"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <a
      href="http://www.grafana.com"
    >
      Link with tooltip
    </a>
  </div>
</div>
`;
